core/runtime/build.rs

```diff
@@ -0,0 +1,56 @@
+use std::fs::copy;
+use std::path::PathBuf;
+
+// auto_layout() identifies the correct linker scripts to use based on the
+// LIBTOCK_PLATFORM environment variable, and copies the linker scripts into
+// OUT_DIR. The cargo invocation must pass -C link-arg=-Tlayout.ld to rustc
+// (using the rustflags cargo config).
+#[cfg(not(feature = "no_auto_layout"))]
+fn auto_layout(out_dir: &str) {
+    const PLATFORM_CFG_VAR: &str = "LIBTOCK_PLATFORM";
+    const LAYOUT_GENERIC_FILENAME: &str = "layout_generic.ld";
+
+    // Note: we need to print these rerun-if commands before using the variable
+    // or file, so that if the build script fails cargo knows when to re-run it.
+    println!("cargo:rerun-if-env-changed={}", PLATFORM_CFG_VAR);
+
+    // Read configuration from environment variables.
+
+    // Read the platform environment variable as a String (our platform names
+    // should all be valid UTF-8).
+    let platform = std::env::var(PLATFORM_CFG_VAR).expect("Please specify LIBTOCK_PLATFORM");
+
+    // Copy the platform-specific layout file into OUT_DIR.
+    let platform_filename = format!("{}.ld", platform);
+    let platform_path: PathBuf = ["layouts", &platform_filename].iter().collect();
+    println!("cargo:rerun-if-changed={}", platform_path.display());
+    if !platform_path.exists() {
+        panic!("Unknown platform {}", platform);
+    }
```

Comment:
Can this be an `assert!`? (Worth doing elsewhere you have a panic in an if.)

Reply:
Yes, it can be. I replaced this pattern in 3 places with `assert!`.